cmd/conformance-tester/pkg/clients/client_kube.go

```diff
@@ -200,75 +240,60 @@ func (c *kubeClient) CreateNodeDeployments(ctx context.Context, log *zap.Sugared
 		return nil
 	}
 
-	log.Info("Preparing MachineDeployments")
+	c.log(log).Info("Preparing MachineDeployments...")
 
 	var mds []clusterv1alpha1.MachineDeployment
-	if err := wait.PollImmediate(3*time.Second, time.Minute, func() (bool, error) {
-		var err error
-		mds, err = scenario.MachineDeployments(ctx, nodeCount, c.opts.Secrets, cluster)
-		if err != nil {
-			log.Warnw("Getting NodeDeployments from scenario failed", zap.Error(err))
-			return false, nil
-		}
-		return true, nil
+	if err := wait.PollImmediate(ctx, 3*time.Second, time.Minute, func() (transient error, terminal error) {
+		mds, transient = scenario.MachineDeployments(ctx, nodeCount, c.opts.Secrets, cluster)
+		return transient, nil
 	}); err != nil {
 		return fmt.Errorf("didn't get NodeDeployments from scenario within a minute: %w", err)
 	}
 
-	log.Info("Creating MachineDeployments...")
+	c.log(log).Info("Creating MachineDeployments...")
 	for _, md := range mds {
-		if err := userClusterClient.Create(ctx, &md); err != nil {
-			return fmt.Errorf("failed to create MachineDeployment: %w", err)
+		if err := wait.PollImmediateLog(ctx, log, 5*time.Second, time.Minute, func() (error, error) {
+			return userClusterClient.Create(ctx, &md), nil
+		}); err != nil {
+			return fmt.Errorf("didn't get NodeDeployments from scenario within a minute: %w", err)
```

Comment:
This error message doesn't look correct (copy-paste from the `wait.PollImmediate` above?).